aslam_cv_cameras/src/ncamera.cc

```diff
@@ -185,136 +239,87 @@ int NCamera::getCameraIndex(const CameraId& id) const {
-  std::vector<aslam::Camera::Ptr> cameras;
-  cameras.push_back(aslam::PinholeCamera::createTestCamera());
-  cameras.push_back(aslam::PinholeCamera::createTestCamera());
-  cameras.push_back(aslam::PinholeCamera::createTestCamera());
-  cameras.push_back(aslam::PinholeCamera::createTestCamera());
-  aslam::NCameraId rig_id;
-  rig_id.randomize();
-  // This defines an artificial camera system similar to the one on the V-Charge or JanETH car.
-  aslam::Position3D t_B_C0(2.0, 0.0, 0.0);
-  Eigen::Matrix3d R_B_C0 = Eigen::Matrix3d::Zero();
-  R_B_C0(1, 0) = -1.0;
-  R_B_C0(2, 1) = -1.0;
-  R_B_C0(0, 2) = 1.0;
-  aslam::Quaternion q_B_C0(R_B_C0);
-  aslam::Position3D t_B_C1(0.0, 1.0, 0.0);
-  Eigen::Matrix3d R_B_C1 = Eigen::Matrix3d::Zero();
-  R_B_C1(0, 0) = 1.0;
-  R_B_C1(2, 1) = -1.0;
-  R_B_C1(1, 2) = 1.0;
-  aslam::Quaternion q_B_C1(R_B_C1);
-  aslam::Position3D t_B_C2(-1.0, 0.0, 0.0);
-  Eigen::Matrix3d R_B_C2 = Eigen::Matrix3d::Zero();
-  R_B_C2(1, 0) = 1.0;
-  R_B_C2(2, 1) = -1.0;
-  R_B_C2(0, 2) = -1.0;
-  aslam::Quaternion q_B_C2(R_B_C2);
-  aslam::Position3D t_B_C3(0.0, -1.0, 0.0);
-  Eigen::Matrix3d R_B_C3 = Eigen::Matrix3d::Zero();
-  R_B_C3(0, 0) = -1.0;
-  R_B_C3(2, 1) = -1.0;
-  R_B_C3(1, 2) = -1.0;
-  aslam::Quaternion q_B_C3(R_B_C3);
-  aslam::TransformationVector rig_transformations;
-  rig_transformations.emplace_back(q_B_C0.inverse(), -t_B_C0);
-  rig_transformations.emplace_back(q_B_C1.inverse(), -t_B_C1);
-  rig_transformations.emplace_back(q_B_C2.inverse(), -t_B_C2);
-  rig_transformations.emplace_back(q_B_C3.inverse(), -t_B_C3);
-  std::string label = "Artificial Planar 4-Pinhole-Camera-Rig";
-  return aligned_shared<aslam::NCamera>(
-      rig_id, rig_transformations, cameras, label);
-}
-
 aslam::NCamera::Ptr NCamera::cloneRigWithoutDistortion() const {
   aslam::NCamera::Ptr rig_without_distortion(this->clone());
   // Remove distortion and assign new IDs to the rig and all cameras.
   for (Camera::Ptr& camera : rig_without_distortion->cameras_) {
     camera->removeDistortion();
-    camera->setId(aslam::CameraId::Random());
+    aslam::CameraId cam_id;
+    generateId(&cam_id);
+    camera->setId(cam_id);
   }
-  rig_without_distortion->setId(aslam::NCameraId::Random());
+
+  aslam::NCameraId ncam_id;
+  generateId(&ncam_id);
+  rig_without_distortion->setId(ncam_id);
   return rig_without_distortion;
 }
 
-bool NCamera::operator==(const NCamera& other) const {
-  bool same = true;
-  same &= getNumCameras() == other.getNumCameras();
-  same &= label_ == other.label_;
-  same &= id_ == other.id_;
-  if (same) {
-    for (size_t i = 0; i < getNumCameras(); ++i) {
-      same &= aslam::checkSharedEqual(cameras_[i], other.cameras_[i]);
-      same &= ((T_C_B_[i].getTransformationMatrix() - other.T_C_B_[i].getTransformationMatrix())
-          .cwiseAbs().maxCoeff() < common::macros::kEpsilon);
+bool NCamera::isValidImpl() const {
+  for (const aslam::Camera::Ptr& camera : cameras_) {
```

Comment:
Seems overly complicated for this simple case?

Reply:
both are fine IMO. I always prefer STL algorithms since they actually might have a better performance than a regular for loop. For me it is completely ok if we keep it like this.